src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2499,14 +2554,23 @@ export class StudyViewPageStore {
         default: []
     });
 
-    readonly mutatedGeneData = remoteData<MutatedGenesData>({
-        await: () => [this.mutationProfiles],
+    readonly mutatedGeneData = remoteData<MutationCountByGeneWithCancerGene[]>({
+        await: () => [this.mutationProfiles, this.oncokbCancerGenes, this.oncokbGenes],
         invoke: async () => {
             if (!_.isEmpty(this.mutationProfiles.result!)) {
                 // TODO: get data for all profiles
-                return internalClient.fetchMutatedGenesUsingPOST({
+                let mutatedGenes = await internalClient.fetchMutatedGenesUsingPOST({
                     studyViewFilter: this.filters
                 });
+                return mutatedGenes.map(item => {
+                    return {
+                        ...item,
+                        oncokbAnnotated: this.oncokbAnnotatedGeneEntrezGeneIds.includes(item.entrezGeneId),
```

Comment:
@adamabeshouse @zhx828 i know i always harp on this and it hasn't seemed to matter much, but it always bothers me when i see a computed which is just a filtering of mobxpromise.  this is a good example because this promise awaits oncokbCancerGenes, but then uses oncokbAnnotatedGeneIds which is a computation of the former.   It would be more straightforward if oncokbTsgEntrezGeneIds was just a utility function that accepted oncokbGenes.   But I know people like to encapsulate stuff.   I wonder if there is a way for mobxpromises to expose methods that would their results.   Just food for thought.

Reply:
I guess you do get a benefit @computed being cached. So it will only compute once per oncokbgenes result.  That's a pretty big win.  But I think the same could be accomplished with @memoized decorator on a utility function.

Reply:
> I wonder if there is a way for mobxpromises to expose methods that would their results.
 
would what their results?

Reply:
also I agree that those should be mobxpromise

Reply:
filter their results.   so, instead of always wrapping mobxpromise with mobxpromise that only filters/manipulates result, the original promise could somehow have methods that filter the result property.     like,  
`  if (mpromise.isComplete) {
      console.log(mpromise.filteredResult)  

}`
where filteredResult is a function defined in mobxpromise declaration that operates on result).

Reply:
I dont like that promise X needs to anticipate how it would be used.

Maybe another solution would be to add a nice factory method for mobxpromises like this:

```
readonly Y = filterMobxPromise(X, result=>{ // return filtered version of result; });

function filterMobxPromise<R,F>(promise:MobxPromise<R>, filter:(result:R)=>F) {
    return remoteData({
        await:()=>[promise],
        invoke:()=>Promise.resolve(filter(promise.result!))
    });
}

```